server/service/integration_vpp_install_test.go

```diff
@@ -1533,3 +1540,147 @@ func (s *integrationMDMTestSuite) TestInHouseAppInstall() {
 	require.Equal(t, "ios", st.SoftwareTitle.SoftwarePackage.Platform)
 	require.WithinDuration(t, time.Now(), st.SoftwareTitle.SoftwarePackage.UploadedAt, time.Hour)
 }
+
+func (s *integrationMDMTestSuite) TestInHouseAppSelfInstall() {
+	t := s.T()
+	s.setSkipWorkerJobs(t)
+	ctx := context.Background()
+
+	// Enroll iPhone
+	iosHost, iosDevice := s.createAppleMobileHostThenEnrollMDM("ios")
+	s.appleVPPConfigSrvConfig.SerialNumbers = append(s.appleVPPConfigSrvConfig.SerialNumbers, iosDevice.SerialNumber)
+
+	// Upload in-house app for iOS, not available in self-service for now
+	s.uploadSoftwareInstaller(t, &fleet.UploadSoftwareInstallerPayload{Filename: "ipa_test.ipa"}, http.StatusOK, "")
+
+	// get its title ID
+	var resp listSoftwareTitlesResponse
+	s.DoJSON("GET", "/api/latest/fleet/software/titles", listSoftwareTitlesRequest{
+		SoftwareTitleListOptions: fleet.SoftwareTitleListOptions{Platform: "ios"},
+	}, http.StatusOK, &resp, "team_id", "0")
+	require.Len(t, resp.SoftwareTitles, 1)
+	require.Equal(t, "ipa_test", resp.SoftwareTitles[0].Name)
+	titleID := resp.SoftwareTitles[0].ID
+
+	activityData := fmt.Sprintf(`{"software_title": "ipa_test", "software_package": "ipa_test.ipa", "team_name": null,
+		"team_id": null, "self_service": false, "software_title_id": %d}`, titleID)
+	s.lastActivityMatches(fleet.ActivityTypeAddedSoftware{}.ActivityName(), activityData, 0)
+
+	// TODO(mna): until iDevice SCEP-based auth is implemented, cheat by adding a
+	// token associated with the iOS host so we can use the self-install endpoint.
+	mysql.ExecAdhocSQL(t, s.ds, func(q sqlx.ExtContext) error {
+		_, err := q.ExecContext(ctx, `INSERT INTO host_device_auth (host_id, token) VALUES (?, 'secret')`, iosHost.ID)
+		return err
+	})
```

Comment:
Same here @cdcme , once we have the cert-based auth we should update this hack to use the proper auth in integration tests.